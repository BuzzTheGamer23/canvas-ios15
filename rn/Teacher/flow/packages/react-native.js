// @flow
export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}
