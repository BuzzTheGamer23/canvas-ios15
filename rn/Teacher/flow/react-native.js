//
// Copyright (C) 2017-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow
/* eslint-disable */

// define basic style prop types
// the internal ones are the best we have
import {
  type ____ViewStyleProp_Internal,
  type ____TextStyleProp_Internal,
  type ____ImageStyleProp_Internal,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes'

declare type ViewStyleProp = ____ViewStyleProp_Internal
declare type TextStyleProp = ____TextStyleProp_Internal
declare type ImageStyleProp = ____ImageStyleProp_Internal
