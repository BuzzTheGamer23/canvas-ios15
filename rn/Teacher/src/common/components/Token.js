// @flow

import React from 'react'
import {
  Text,
  StyleSheet,
} from 'react-native'

type TokenProps = {
  +style?: any,
  +children?: string,
  +color: string,
}

const Token = (props: TokenProps): * => {
  let {
    style,
    children,
    color,
  } = props

  return (
    <Text {...props} style={[styles.token, { color, borderColor: color }, style]}>
      {children && children.toUpperCase()}
    </Text>
  )
}

const styles = StyleSheet.create({
  token: {
    fontSize: 11,
    fontWeight: '600',
    borderRadius: 9,
    borderWidth: 1,
    backgroundColor: 'white',
    paddingTop: 3,
    paddingBottom: 0,
    paddingLeft: 6,
    paddingRight: 6,
    overflow: 'hidden',
  },
})

export default Token
