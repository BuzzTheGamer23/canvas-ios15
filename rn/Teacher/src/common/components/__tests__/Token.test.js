// @flow

import 'react-native'
import React from 'react'
import Token from '../Token'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

test('Token renders correctly', () => {
  let tree = renderer.create(
    <Token color='#0077FF'>Shiny Buckles</Token>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
