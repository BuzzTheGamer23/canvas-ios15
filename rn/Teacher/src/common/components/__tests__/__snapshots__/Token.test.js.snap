// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Token renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  color="#0077FF"
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderRadius": 9,
        "borderWidth": 1,
        "fontSize": 11,
        "fontWeight": "600",
        "overflow": "hidden",
        "paddingBottom": 0,
        "paddingLeft": 6,
        "paddingRight": 6,
        "paddingTop": 3,
      },
      Object {
        "borderColor": "#0077FF",
        "color": "#0077FF",
      },
      undefined,
    ]
  }
>
  SHINY BUCKLES
</Text>
`;
